bridges/sloghandler/handler.go

```diff
@@ -0,0 +1,360 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+// Package sloghandler provides a bridge between the [log/slog] and
+// OpenTelemetry logging.
+package sloghandler // import "go.opentelemetry.io/contrib/bridges/sloghandler"
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"slices"
+	"sync"
+
+	"go.opentelemetry.io/otel/log"
+	"go.opentelemetry.io/otel/log/noop"
+)
+
+const (
+	bridgeName = "go.opentelemetry.io/contrib/bridge/sloghandler"
+	// TODO: hook this into the release pipeline.
+	bridgeVersion = "0.0.1-alpha"
+)
+
+type config struct{}
+
+// Option configures a [Handler].
+type Option interface {
+	apply(config) config
+}
+
+// Handler is an [slog.Handler] that sends all logging records it receives to
+// OpenTelemetry.
+type Handler struct {
+	// Ensure forward compatibility by explicitly making this not comparable.
+	noCmp [0]func() //nolint: unused  // This is indeed used.
+
+	attrs  *kvBuffer
+	group  *group
+	logger log.Logger
+}
+
+// Compile-time check *Handler implements slog.Handler.
+var _ slog.Handler = (*Handler)(nil)
+
+// New returns a new [Handler] to be used as an [slog.Handler].
+func New(lp log.LoggerProvider, _ ...Option) *Handler {
+	if lp == nil {
+		// Do not panic.
+		lp = noop.NewLoggerProvider()
+	}
+	return &Handler{
+		logger: lp.Logger(
+			bridgeName,
+			log.WithInstrumentationVersion(bridgeVersion),
+		),
```

Comment:
These need to likely be the package/module name, not the bridge name: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/bridge-api.md#get-a-logger

The bridge name and version could be added as an attribute here. More investigation is required.

Reply:
Maybe the bridge should accept `log.Logger` instead of `log.LoggerProvider`? Then the user would have full control what is the instrumentation scope.

Reply:
Switched to accepting a logger.